FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/DataProcessor/ParquetDataProcessor.cs

```diff
@@ -76,7 +82,20 @@ public sealed class ParquetDataProcessor : IColumnDataProcessor
                 throw new ParquetDataProcessorException($"The FHIR schema node could not be found for schema type '{processParameters.SchemaType}'.");
             }
 
-            var inputStream = ConvertJsonDataToStream(processParameters.SchemaType, preprocessedData.Values);
+            // Convert data based on schema
+            JsonBatchData processedData;
+            if (IsCustomizedSchemaType(processParameters.SchemaType))
+            {
+                var rawResourceType = processParameters.SchemaType
+                    .Substring(0, processParameters.SchemaType.Length - FhirParquetSchemaConstants.CustomizedSchemaSuffix.Length);
+                processedData = _fhirDataConverter.Convert(inputData, rawResourceType, cancellationToken);
+            }
+            else
+            {
+                processedData = _defaultDataConverter.Convert(inputData, schema, cancellationToken);
```

Comment:
Seems we can make the default and custom schema converters to a common interface IDataSchemaConverter/IDataSchemaTransformer which consumes `inputData`, `schemaType` and `cancellationToken` to help transforms data. And the schema transformation part can be seperated from ParquetDataProcessor (should focus on transform json to parquets?)

Reply:
Sure, updated.